near-sdk-macros/src/core_impl/info_extractor/attr_sig_info.rs

```diff
@@ -105,7 +105,7 @@ impl AttrSigInfo {
 
         if let Some(ref receiver) = receiver {
             if matches!(method_type, MethodType::Regular) {
-                if receiver.mutability.is_none() {
+                if receiver.mutability.is_none() || receiver.reference.is_none() {
```

Comment:
I'm not sure why we can't just have
```rust
if receiver.reference.is_none() {
```
but `compilation_tests/payable_view.rs` successfully compiles even though its expected not to.

This conditional should just only be checking for `mut self` and `self` signatures, where checking for mutability shouldn't matter at all. But somehow also related to `pay(&self)` signature in `payable_vew.rs`? Maybe there's some sort of complexity somewhere we're not noticing or I'm probably missing something here.

Reply:
Nah this is intentional. This was the point of the PR that `mut self` was treated as mutating state and tries to write the data after. If you check only for reference, you don't differentiate between mutable and immutable references